### DISCUSSION TITLE
Where can I get historical quote data?

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#579] [Q&A] [answered]
There are many places to get stock market data.  Check with your brokerage or other commercial sites.
If you're looking for a free developer API source, here's a few _(in no particular order)_:

- [Alpaca Markets](https://alpaca.markets/data)
- [Alpha Vantage](https://www.alphavantage.co)
- [FinancialData.net](https://financialdata.net/pricing?via=facioquo) - use promo code `FACIOQUO10` for 10% discount on paid tiers[^1]
- [FinnHub.io](https://finnhub.io)
- [iexcloud.io](https://iexcloud.io)
- [Polygon.io](https://polygon.io)
- [TwelveData](https://twelvedata.com)

[^1]: We are not affiliated with FinancialData.net and make no warranty on their service.  By using the `FACIOQUO10` promo code, you get a 10% discount.  Our open-source [FacioQuo](https://github.com/facioquo) community organization benefits when you use the [referral link](https://financialdata.net/pricing?via=facioquo) when subscribing to a paid tier.

**Comment by DaveSkender** [comment]
If you're looking for some C# packaged APIs for the cryptocurrency world, some of which may provide historical data, @JKorf has been building a bunch here: https://github.com/JKorf/

**Reply by flupkede** [reply]
I would like to add iexcloud.io , certainly if you are interested in European market.

**Comment by Dave-Wagner** [comment]
Another potential source of data is the MetaTrader 5 terminal itself through their Python package MetaTrader5, for those using that platform anyway. The major advantage is that you'll be working with the brokers' data that you'll potentially be trading on.

**Comment by DaveSkender** [comment, marked as answer]
See the above list of professionally managed data sources.  Most have a freemium model to support developers and hobbyists.
Feel free to add your favorites or other API helpers in the discussion area below.

**Comment by jotenwinds** [comment]
There is another good and free web site to download historical data (US, World, ...). No API, but there are .CSV file easily to read and load => https://stooq.com/db/h/

**Comment by ooples** [comment]
@DaveSkender I would like to suggest my newest API library to be added to this list. I created a wrapper API for Yahoo Finance that can get you any information you are looking for. I currently support getting top trending stocks, historical price data, dividends, stock splits, capital gains, recommended stocks, and I have many other items I will be adding in the next few days.

https://github.com/ooples/OoplesFinance.YahooFinanceAPI

**Comment by martonb** [comment]
For anyone interested in Crypto data, I used this: https://www.cryptodatadownload.com/data
Lately, it needs a free account signup, to download, though...

**Comment by DaveSkender** [comment]
https://analyzingalpha.com/best-market-data-apis

**Comment by DaveSkender** [comment]
Anybody have any success using the Robinhood API lately?

**Comment by DaveSkender** [comment]
I've not had a chance to try it out yet, but this new data provider [FinancialData.net](https://financialdata.net/pricing?via=facioquo) reached out to me and is offering 10% discounts for our users.  See note in top-level discussion for more information.  They also have a free tier that'd work for general development purposes.
